src/ui/app/pages/send.jsx

```diff
@@ -895,20 +896,20 @@ const AddressPopup = ({
   }, [txInfo]);
 
   const handleInput = async (e) => {
-    const val = e.target.value;
+    const value = e.target.value;
```

Comment:
Avoid using `e.target.value` directly in async operations to prevent race conditions. If the user types in more text before the operation completes, the function will use the new text instead of the original input, leading to an inaccurate display value.

```
var target = {
  value: 'foo'
}

var watcher = target.value

target.value = 'bar'

console.log('watcher', watcher, 'target.value', target.value) // watcher foo target.value bar

```

Reply:
I don't quite understand the concern: if user types more text, this function will be called with a new/separate `e` object :thinking:

Reply:
@mkazlauskas but in the case of the race condition, request A finishing after request B, event.target references the element so it will have the latest value and not value during event A when requets A finaly resolves.

Reply:
Got it, so this isn't a change request? It's already saving the input value in function scope before any async calls

Reply:
It's saving at the begning but it would reference e.target.value again before returning.

Reply:
https://github.com/input-output-hk/nami/blob/9d21c1a6e9c4213f7579981407866ce0a14a3b13/src/ui/app/pages/send.jsx#L949